tools/analyzer_plugin/test/unit/util/boilerplate_utils_test.dart

```diff
@@ -0,0 +1,282 @@
+          expect(editList.first.replacement, 'part \'foo.over_react.g.dart\';\n\n');
+        });
+
+        test('when there is an existing part directive in the file', () async {
+          final result = await parseAndGetResolvedUnit(sourceWithNonOverReactPart, path: 'foo.dart');
+          final sourceChange = await buildFileEdit(result, (builder) {
+            addOverReactGeneratedPartDirective(builder, result.unit, result.uri);
+          });
+          final editList = sourceChange.edits?.firstOrNull?.edits;
+
+          expect(editList, isNotNull);
+          expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+          final offset = result.unit.lineInfo.getOffsetOfLineAfter(result.unit.directives.last.end);
+          expect(editList.first.offset, offset, reason: 'new part should be on the line after existing part');
+          expect(editList.first.length, 0, reason: 'nothing is replaced');
+          expect(editList.first.replacement, 'part \'foo.over_react.g.dart\';\n\n');
+        });
+      });
+
+      test('replaces existing over_react part directive if it doesn\'t match file uri', () async {
+        final result = await parseAndGetResolvedUnit(sourceWithOverReactPart, path: 'different_file_name.dart');
+        final sourceChange = await buildFileEdit(result, (builder) {
+          addOverReactGeneratedPartDirective(builder, result.unit, result.uri);
+        });
+        final editList = sourceChange.edits?.firstOrNull?.edits;
+
+        expect(editList, isNotNull);
+        expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+        final part = getOverReactGeneratedPartDirective(result.unit);
+        expect(editList.first.offset, part.offset, reason: 'new part should be on the line after existing part');
+        expect(editList.first.length, part.length, reason: 'mismatched part is replaced');
+        expect(editList.first.replacement, 'part \'different_file_name.over_react.g.dart\';');
+      });
+    });
+
+    group('removeOverReactGeneratedPartDirective', () {
+      group('does nothing if', () {
+        test('there is no part directive in file', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNoPart,
+            (builder, result) => removeOverReactGeneratedPartDirective(builder, result.unit),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+
+        test('there is no over_react part directive in the file', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNonOverReactPart,
+            (builder, result) => removeOverReactGeneratedPartDirective(builder, result.unit),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+      });
+
+      test('removes over_react part directive if it exists', () async {
```

Comment:
Since we've tested how the utilities react when adding a part when one already exists, it may be worth testing how the utilities react when removing the over_react part when another one exists

Reply:
Good idea! I'll add that test case.